### PR TITLE
Add parser for TOML

### PR DESCRIPTION
**Pull request by iggy890**
This PR adds `tree-sitter-toml` parser to `CodeEditLanguages`. NOTE: The parser is a fork.

### CODE REVIEW

File: CodeLanguages-Container/CodeLanguages-Container.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -6,74 +6,11 @@
 	objectVersion = 56;
 	objects = {
 
-/* Begin PBXBuildFile section */
```
Comment (lukepistrol) [inline CodeLanguages-Container/CodeLanguages-Container.xcodeproj/project.pbxproj:9]: There's a lot of deletions in the `project.pbxproj` file. Please make sure, that you correctly resolved all dependencies. There shouldn't be any deletions at all - just additions related to the `tree-sitter-toml` repo. Take a look at this PR for example: https://github.com/CodeEditApp/CodeEditLanguages/pull/28/files#diff-0a20baec64c8334a037d370407084a60dda04a6bee936c7eb1608656b26748cbR8-R684

**Review by lukepistrol — CHANGES_REQUESTED** [review]
One thing about the `project.pbxproj` file you need to investigate a bit further.

**Review by lukepistrol — CHANGES_REQUESTED** [review]
Also it seems that the `highlights.scm` which is present in your fork of `tree-sitter-toml` [here](https://github.com/iggy890/tree-sitter-toml/blob/feature/spm/queries/highlights.scm), is still missing. Have you missed adding it to the commit somehow? It should automatically be copied over into the `Sources/CodeEditLanguages/Resources/tree-sitter-toml` folder when running the `build_framework.sh`. Please investigate this issue as well. And make sure the script adds this automatically (don't just copy it over, since it will be missing the next time the script is run).

**EDIT:** Forgot to fill in the link above. Fixed now.

File: CodeLanguages-Container/CodeLanguages-Container.xcodeproj/project.pbxproj (hunk 1 above)
Comment (iggy890) [reply CodeLanguages-Container/CodeLanguages-Container.xcodeproj/project.pbxproj:9]: That's weird I don't remember deleting anything, I'll look into it

**Comment by lukepistrol** [comment]
Also the package does not build as you can see [here](https://github.com/CodeEditApp/CodeEditLanguages/actions/runs/4270784251/jobs/7435074659)

File: CodeLanguages-Container/CodeLanguages-Container.xcodeproj/project.pbxproj (hunk 1 above)
Comment (lukepistrol) [reply CodeLanguages-Container/CodeLanguages-Container.xcodeproj/project.pbxproj:9]: You somehow deleted the framework target.

Screenshot 2023-02-25 at 18 00 57

And that's what it should look like:

Screenshot 2023-02-25 at 18 02 09

Comment (iggy890) [reply CodeLanguages-Container/CodeLanguages-Container.xcodeproj/project.pbxproj:9]: How did I do that? I'll try to restore that

**Comment by iggy890** [comment]
> Also it seems that the `highlights.scm` which is present in your fork of `tree-sitter-toml` here, is still missing. 

I've re-ran `./build_framework.sh` with the following output:
shot
Why this doesn't add tree-sitter-toml to `Resources` is unknown to me.

File: CodeLanguages-Container/CodeLanguages-Container.xcodeproj/project.pbxproj (hunk 1 above)
Comment (lukepistrol) [reply CodeLanguages-Container/CodeLanguages-Container.xcodeproj/project.pbxproj:9]: Therefore when running the `build_framework.sh` script, it won't succeed. When running it with the `--debug` flag you get the error `xcodebuild: error: The project named "CodeLanguages-Container" does not contain a scheme named "CodeLanguages-Container". The "-list" option can be used to find the names of the schemes in the project.`

**Comment by iggy890** [comment]
> Also it seems that the `highlights.scm` which is present in your fork of `tree-sitter-toml` here, is still missing. 

I've re-ran `./build_framework.sh` with the following output:
shot
Weirdly, it doesn't add anything to `Resources/`

**Comment by lukepistrol** [comment]
> I've re-ran ./build_framework.sh with the following output:

@iggy890 running the script on your fork gives me this output not the one you posted

Screenshot 2023-02-25 at 18 08 50

**Comment by iggy890** [comment]
Yeah, it only runs in the macOS terminal not the VSCode one.

**Comment by iggy890** [comment]
Everything should be restored :)

**Comment by lukepistrol** [comment]
Ok, now please update your branch to the current main and rebuild the framework once again since in the mean time the `SQL parser` was merged.

**Comment by iggy890** [comment]
Ok, I will add the SQL parser

**Comment by lukepistrol** [comment]
> Ok, I will add the SQL parser

It's already added. Just update your branch with the latest `main`.

**Comment by lukepistrol** [comment]
@iggy890 You're not supposed to add the sql parser again as it is already merged. Please just update your branch (e.g. rebase) to contain the changes of the `main` branch of this repo..

**Comment by iggy890** [comment]
BTW: Even after updating my branch there are still merge conflicts.

**Comment by lukepistrol** [comment]
> BTW: Even after updating my branch there are still merge conflicts.

As you can see here, your branch is not up-to-date with `main` please do a rebase on CodeEditLanguages' `main` branch.

Screenshot 2023-02-25 at 22 34 27

**Pull request closed without merging**